### ISSUE TITLE
correctly retrigger builds when project wide required package is changed

### ISSUE THREAD

**Issue by JanZerebecki** [openSUSE/open-build-service#8401]
# Issue Description
When a project wide required package is changed it seems that it triggers no rebuild in reverse dependencies. Correctly retrigger affected builds.

Expected Result
============
Correctly retrigger affected builds.

How to Reproduce
===============
1. Have a project with a prjconf that has a required in a type:kiwi, like: `Required: containment-rpm-foo`.
2. Have a kiwi package. E.g. a product CD.
3. Change such a project wide required package.
4. See that no rebuild is triggered for the kiwi package but all repos reach published state. But the changed version can be seen in buildinfo for the kiwi package.

Further Information
===============
* A link to an OBS instance showing the issue: https://build.suse.de/project/show/Devel:Cloud:9:Appliances

**Comment by coolo** [comment]
I have the same issue with product-builder home:coolo:branches:SUSE:SLE-15-SP2:GA - and it's really painful

**Comment by mlschroe** [comment]
I'm not sure I understand this. Is there a package in `Devel:Cloud:9:Appliances` that shows this behavior?

**Comment by JanZerebecki** [comment]
@mlschroe yes, which step did you have problems following or matching to `Devel:Cloud:9:Appliances`?

**Comment by morgwai** [comment]
I think I've just encountered this bug in my [Debian kernel project](https://build.opensuse.org/project/show/home:morgwai:tbfive-kernel): I've added `Required: linux-libc-dev` to [the config](https://build.opensuse.org/projects/home:morgwai:tbfive-kernel/prjconf), yet no rebuild was triggered.
Moreover, after triggering manually, it still fails due to the lack of `linux-libc-dev` package:
```
[   44s] [300/527] installing libc6-dev-2.41-12+deb13u3
[   44s] dpkg: libc6-dev:amd64: dependency problems, but configuring anyway as you requested:
[   44s]  libc6-dev:amd64 depends on linux-libc-dev; however:
[   44s]   Package linux-libc-dev is not installed.
```
(...)
```
[  648s] /usr/bin/make -f /usr/src/packages/BUILD/scripts/Makefile.build obj=scripts scripts/unifdef
[  648s] # HOSTCC  scripts/unifdef
[  648s]   gcc -Wp,-MMD,scripts/.unifdef.d -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 -fomit-frame-pointer -std=gnu11   -I /usr/src/packages/BUILD/scripts/include   -I scripts   -o scripts/unifdef /usr/src/packages/BUILD/scripts/unifdef.c   
[  648s] In file included from /usr/include/errno.h:28,
[  648s]                  from /usr/src/packages/BUILD/scripts/unifdef.c:51:
[  648s] /usr/include/x86_64-linux-gnu/bits/errno.h:26:11: fatal error: linux/errno.h: No such file or directory
```

**Comment by morgwai** [comment]
So the lack of `linux-libc-dev` was resolved using `Keep:` directive (kudos to [this post](https://forums.opensuse.org/t/linux-errno-h-no-such-file-or-directory-when-building-kernel-for-debian-13/194407/7)).
However even after adding `Keep:`, which changed the list of packages installed on the environment and eventually led to a successful build, a rebuild was not triggered automatically. Should I file another bug regarding triggering via `Keep:` ?

**Comment by mlschroe** [comment]
Please do not use `keep`, it's an obsolete mechanism. If your package really requires itself for building, it should declare this in Build-Depends.

**Comment by morgwai** [comment]
The package is based on [the official Debian kernel packages](https://salsa.debian.org/morgwai/linux/-/commits/b7.1-tbfive1), I've just added 2 patches related to TB5/eGPU handling, `Build-Depends:` was definitely not changed from the original. Furthermore, I've just verified that its build _DOES_ depend on `linux-libc-dev` via several transitive dependencies of packages listed in `Build-Depends:`, yet OBS refuses to install it without `Keep:` (I tried `Required:` before and it didn't work). From what you wrote, it seems this is not the expected behavior, so I've filed #19830
Thanks!
